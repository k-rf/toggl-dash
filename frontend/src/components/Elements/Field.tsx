import { TextField as MuiTextField, TextFieldProps as MuiTextFieldProps } from "@mui/material";

type Props = Pick<MuiTextFieldProps, "label" | "error">;

export const Field = (props: Props) => {
  return <MuiTextField {...props} />;
};
