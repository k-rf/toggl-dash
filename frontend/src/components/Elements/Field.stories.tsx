import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import { Field } from "./Field";

type Meta = ComponentMeta<typeof Field>;
type Story = ComponentStoryObj<typeof Field>;

export default { title: "Elements/Field", component: Field } as Meta;

export const Default: Story = {};

export const Label: Story = {
  ...Default,
  args: {
    ...Default.args,
    label: "ラベル",
  },
};

export const Error: Story = {
  ...Default,
  args: {
    ...Default.args,
    label: "エラー",
    error: true,
  },
};
