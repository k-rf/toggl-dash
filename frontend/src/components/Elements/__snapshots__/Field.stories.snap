// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements/Field > Default 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Elements/Field > Error 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor=":r2:"
    id=":r2:-label"
  >
    エラー
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r2:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1ftyaf0"
      >
        <span>
          エラー
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Elements/Field > Label 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor=":r1:"
    id=":r1:-label"
  >
    ラベル
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r1:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1ftyaf0"
      >
        <span>
          ラベル
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
